shadowtools/tests/test_shadow_exec.py

```diff
@@ -58,7 +58,10 @@ def test_bash_script_sleep(self) -> None:
                 "--",
                 "bash",
                 "-c",
-                "date -Ins; sleep 1.001; date -Ins",
+                # we would use 'sleep' here, but /bin/sleep in newer coreutils seems to add an extra
+                # nanosecond to the specified time:
+                # https://github.com/shadow/shadow/pull/3663#issuecomment-3341062224
+                "date -Ins; coproc read -t 1.001; wait; date -Ins",
```

Comment:
TIL about coproc; clever! It'd be nice if we could keep this less esoteric, but nothing comes to mind that wouldn't add a fair bit of complexity. I'm good with this solution.

Reply:
Yeah, I'm happy if someone changes it later, but I'm not sure of a nicer way.